Plugin/Api/CartSearchRepository.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+
+namespace Klaviyo\Reclaim\Plugin\Api;
+
+
+use Magento\Quote\Api\CartRepositoryInterface;
+use Magento\Quote\Api\Data\CartExtensionFactory;
+use Magento\Quote\Model\QuoteIdToMaskedQuoteId;
+use Magento\Quote\Model\CartSearchResults;
+
+class CartSearchRepository
+{
+
+    const KL_MASKED_ID = 'kl_masked_id';
+
+    /**
+     * QuoteId Masker
+     *
+     * @var QuoteIdToMaskedQuoteId
+     */
+    private $quoteIdMask;
+
+    /**
+     * Cart Extension Attributes Factory
+     *
+     * @var CartExtensionFactory
+     */
+    protected $extensionFactory;
+
+    /**
+     * CartRepositoryPlugin constructor
+     *
+     * @param CartExtensionFactory $extensionFactory
+     * @param QuoteIdToMaskedQuoteId $quoteIdToMaskedQuoteId
+     */
+    public function __construct(
+        CartExtensionFactory $extensionFactory,
+        QuoteIdToMaskedQuoteId $quoteIdToMaskedQuoteId
+    ){
+        $this->extensionFactory = $extensionFactory;
+        $this->quoteIdMask = $quoteIdToMaskedQuoteId;
+    }
+    /**
+     * Add "kl_masked_id" extension attribute to order data object to make it accessible in API data
+     *
+     * @param CartRepositoryInterface $subject
+     * @param CartSearchResults $searchResult
+     *
+     * @return CartSearchResults
+     */
+    public function afterGetList(CartRepositoryInterface $subject, CartSearchResults $searchResult)
+    {
+        $quotes = $searchResult->getItems();
+
+        foreach ($quotes as $quote) {
+            $maskedId = $this->getMaskedIdFromQuoteId($quote->getId());
+            $extensionAttributes = $quote->getExtensionAttributes();
+            $extensionAttributes = $extensionAttributes ? $extensionAttributes : $this->extensionFactory->create();
+            $extensionAttributes->setData(self::KL_MASKED_ID, $maskedId);
+            $quote->setExtensionAttributes($extensionAttributes);
+        }
+
+        return $searchResult;
+    }
+
+    public function getMaskedIdFromQuoteId($quoteId)
+    {
+        try {
+            $quoteId = $this->quoteIdMask->execute($quoteId);
```

Comment:
@remstone7 @klaviyojad 
FYI this change causing ~2x slow down on the quotes list retrieval request, but that's an issue in the Magneto core.
More details here: https://github.com/magento/magento2/pull/33324